base_layer/wallet/src/base_node_service/monitor.rs

```diff
@@ -95,27 +106,37 @@ impl<T: WalletBackend + 'static> BaseNodeMonitor<T> {
 
     async fn monitor_node(&mut self) -> Result<(), BaseNodeMonitorError> {
         loop {
+            let start = Instant::now();
             let mut client = self
                 .wallet_connectivity
                 .obtain_base_node_wallet_rpc_client()
                 .await
                 .ok_or(BaseNodeMonitorError::NodeShuttingDown)?;
+            trace!(
+                target: LOG_TARGET,
+                "Obtain RPC client {} ms",
+                Instant::now().duration_since(start).as_millis(),
+            );
 
             let base_node_id = match self.wallet_connectivity.get_current_base_node_id() {
                 Some(n) => n,
                 None => continue,
             };
 
+            let start = Instant::now();
             let tip_info = client.get_tip_info().await?;
-
             let chain_metadata = tip_info
                 .metadata
                 .ok_or_else(|| BaseNodeMonitorError::InvalidBaseNodeResponse("Tip info no metadata".to_string()))
                 .and_then(|metadata| {
                     ChainMetadata::try_from(metadata).map_err(BaseNodeMonitorError::InvalidBaseNodeResponse)
                 })?;
+            // Note: Do not use an RPC ping call to measure latency here, as the time measured for that always follows
```

Comment:
So my understanding of the RPC ping message is that it is supposed to be a measure of the round trip response without a database read so it is a better measure of network latency that should be divorced from the business of the base node database. If the metadata call and this are linked then that actually points to a problem on the base node that the RPC Pong response is somehow being blocked by Database activity.

Reply:
Really difficult to make sense of this. I could pick this up in the logs, and these may be from the same conversation or not but it is difficult to tell:

``` rust
CW:    00:15:41.694385500 [wallet::base_node_service::chain_metadata_monitor] TRACE Obtain RPC client 0 ms
   BN: 00:15:43.309834400 [comms::rpc] DEBUG [Peer=`ff03297bea5f3f548a85dd72c6`] Got request RequestID=12, Deadline=120s, Flags=(empty), Message=0 byte(s) 
CW:    00:16:56.568149800 [comms::rpc::client] TRACE Received response (64 byte(s)) from request #12 (protocol = t/bnwallet/1, method=5) in 75s
CW:    00:16:56.568175700 [wallet::base_node_service::chain_metadata_monitor] TRACE Obtain metadata from client 74873 ms
CW:    00:16:56.568225000 [comms::rpc::client] DEBUG Ping (protocol t/bnwallet/1) sent in 3.00µs
   BN: 00:16:58.072925300 [comms::rpc] TRACE RPC TIMING(REQ_ID=3): 'transmit' took 53.64s (SLOW)
   BN: 00:16:58.073036500 [comms::rpc] DEBUG [Peer=`ff03297bea5f3f548a85dd72c6` t/bnwallet/1] sending ACK response.
CW:    00:18:23.682751100 [wallet::base_node_service::chain_metadata_monitor] TRACE Obtain latency (ping) from client 87114 ms
   BN: 00:18:25.297436900 [comms::rpc] TRACE RPC TIMING(REQ_ID=3): 'transmit' took 85.17s (SLOW) 
```

What stands out is:
- the wallet measures 74.873s for a round trip metadata request; the base node reports a slow transmit of 53.64s
- the wallet measures 87.114s for a round trip ping request; the base node reports a slow transmit of 85.17s

If those timings are linked it would suggest being not blocked by database activity.